### PR TITLE
feat(components): Add page disable button

### PR DESCRIPTION
**Pull request by konst8**
## Changes
- Add page disable button in the admin menu

## Related Issues
#1055

### CODE REVIEW

File: packages/bodiless-components/src/PageDisableButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+type Data = {
+  disabledPages?: string[],
+};
+
+const useMenuOptions = (): TMenuOption[] => {
+  const { node } = useNode<Data>();
+  const context = useEditContext();
+  const { disabledPages = [] } = node.data;
+  const isPageInDisabledList = disabledPages && disabledPages.indexOf(node.pagePath) > -1;
+  const [isPageDisabled, setPageDisabled] = useState<boolean>(isPageInDisabledList);
```
Comment (VancheeZze) [inline packages/bodiless-components/src/PageDisableButton/index.tsx:90]: @konst8 I think it is ok for now to implement it this way with the local state. Make sure it does not produce a hooks warning. We will keep disabled pages in the context anyway in coming stories and that's where we will refactor this piece and get rid of the local context and `useNode` logic.

File: examples/test-site/gatsby-config.js [hunk 2]
```diff
@@ -15,6 +16,17 @@ require('dotenv').config({
 
 const SITEURL = process.env.SITE_URL;
 
+const getDisabledPages = () => {
+  try {
+    const json = fs.readFileSync('./src/data/site/disabled-pages.json');
+    const data = JSON.parse(json.toString());
+    return data.disabledPages || [];
+  } catch (e) {
```
[the remark this answers is not in the document]
Comment (konst8) [reply examples/test-site/gatsby-config.js]: Added.

File: packages/bodiless-components/src/PageDisableButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+type Data = {
+  disabledPages?: string[],
+};
+
+const useMenuOptions = (): TMenuOption[] => {
+  const { node } = useNode<Data>();
+  const context = useEditContext();
+  const { disabledPages = [] } = node.data;
+  const isPageInDisabledList = disabledPages && disabledPages.indexOf(node.pagePath) > -1;
```
[the remark this answers is not in the document]
Comment (konst8) [reply packages/bodiless-components/src/PageDisableButton/index.tsx]: Done.

File: packages/bodiless-components/src/PageDisableButton/index.tsx (hunk 1 above)
Comment (konst8) [reply packages/bodiless-components/src/PageDisableButton/index.tsx]: Removed local state in favor of observed content node.

File: packages/bodiless-components/src/PageDisableButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+const Form = (props: FormProps) => {
+  const { ComponentFormFieldTitle, ComponentFormTitle } = useMenuOptionUI();
+  const { state: { formTitle, formDescription } } = props;
+  return (
+    <ContextMenuForm {...props}>
+      <ComponentFormTitle>
+        {formTitle}
+      </ComponentFormTitle>
+      <ComponentFormFieldTitle>
+        {formDescription.map(string => <p>{string}</p>)}
```
[the remark this answers is not in the document]
Comment (konst8) [reply packages/bodiless-components/src/PageDisableButton/index.tsx]: Done.

**Review by VancheeZze — APPROVED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by VancheeZze — APPROVED** [review]

**Review by irkhodzka — APPROVED** [review]

**Pull request merged**